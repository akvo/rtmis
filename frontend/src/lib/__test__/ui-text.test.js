import "@testing-library/jest-dom";
import uiText from "../ui-text";

describe("UI Text", () => {
  test("check text languages", () => {
    expect(uiText).toHaveProperty("de");
    expect(uiText).toHaveProperty("en");
    expect(uiText).toMatchSnapshot();
  });
});
