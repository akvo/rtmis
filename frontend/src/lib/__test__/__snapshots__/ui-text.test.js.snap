// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Text check text languages 1`] = `
Object {
  "de": Object {},
  "en": Object {
    "accountDisclaimer": "The user is accountable for his/her account and in case there are any changes (Transfers, retirement, any kind of leave, resignation etc) this should be communicated to the County Administrator or National Super Admin who might be able to assign the roles to the new officer.",
    "addOrganisation": "Add Organization",
    "addUser": "Add User",
    "approvalsTab1": "My Pending",
    "approvalsTab2": "Subordinates Approvals",
    "approvalsTab3": "Approved",
    "backHome": "Back to Homepage",
    "batchDatasets": "Batch Datasets",
    "batchHintDesc": "The operation of merging datasets cannot be undone, and will Create a new batch that will require approval from you admin",
    "batchHintText": "You are about to create a Batch CSV File",
    "batchName": "Batch Name",
    "batchNoApproverMessage": "Can't create batch data, because there's no approver yet.",
    "batchSelectedDatasets": "Batch Selected Datasets",
    "browseComputer": "Browse your computer",
    "bulkUploadNoApproverMessage": "Can't upload data, because there's no approver yet.",
    "ccDescriptionPanel": "Instant access to all the administration pages and overview panels for data approvals.",
    "contactAdmin": "Please contact the administrator",
    "controlCenter": "Control Center",
    "copyright": "© 2022 — Ministry of Health",
    "countdownTitle": "Time remaining to achieve national RUSH targets:",
    "createNewBatch": "Create a new batch",
    "dashboards": "Dashboards",
    "dataDownloadButton": "Download Data",
    "dataDownloadText": <React.Fragment>
      This is where you :
      <ul>
        <li>
          Access downloaded data
        </li>
      </ul>
    </React.Fragment>,
    "dataDownloadTitle": "Data Download",
    "dataExportFail": "Data download failed",
    "dataExportSuccess": "Data downloaded successfully",
    "dataUploadButton": "Data Upload",
    "dataUploadText": <React.Fragment>
      This is where you :
      <ul>
        <li>
          Download upload template
        </li>
        <li>
          Bulk upload new data
        </li>
        <li>
          Bulk update existing data
        </li>
      </ul>
    </React.Fragment>,
    "dataUploadTitle": "Data Upload",
    "day": "Day",
    "deleteOrganisationDesc": [Function],
    "deleteOrganisationTitle": "You are about to delete the organization",
    "deleteUserDesc": "The User will no longer be able to access the RUSH platform as an Enumrator/Admin etc",
    "deleteUserHint": "Deleting this user will not delete the data association(s)",
    "deleteUserTitle": "You are about to delete the user",
    "download": "Download",
    "dropFile": "Drop your file here",
    "editOrganisation": "Edit Organization",
    "editUser": "Edit User",
    "error": "Error",
    "errorAuth": "You are not authorised to access this page",
    "errorDataLoad": "Could not load data",
    "errorFileList": "Could not fetch File list",
    "errorMandatoryFields": "Please answer all the mandatory questions",
    "errorPageNA": "Oops, this page is not available",
    "errorSomething": "Something went wrong",
    "errorURL": "Please check the URL again or let us take you back to the RUSH homepage",
    "errorUnknown": "An unknown error occurred",
    "errorUserLoad": "Failed to load user data",
    "errorVerifyCreds": "Please verify your credentials for the requested resource",
    "existingApproverDescription": "Please update the setup in manage validation tree or remove these forms for the current user",
    "existingApproverTitle": "There are existing approvers for:",
    "failed": "Failed",
    "fetchingForm": "Fetching form..",
    "fileUploadFail": "Could not upload file",
    "fileUploadSuccess": "File uploaded successfully",
    "finish": "Finish",
    "footerAboutDescription": "All the data contained in the RUSH platform is aggregated from both primary and secondary data sources from the 47 counties in Kenya. The data is updated on a monthly basis.",
    "footerAboutTitle": "About Data",
    "footerContactAddress": "Afya House, Cathedral Road, P.O. Box:30016–00100, Nairobi, Kenya",
    "footerContactEmail": "xxx@health.go.ke",
    "footerContactFeedback": Object {
      "text": "Feedback Form",
      "url": "https://docs.google.com/forms/d/e/1FAIpQLSf5rjD66CCcMWYgFwkNp8Xb1lRJoec1CwhhPnjOd-mU84ktPA/viewform",
    },
    "footerContactPhone": "+25x-xx-xxxxx",
    "footerContactTitle": "Contacts",
    "footerExternalLinkItems": Array [
      Object {
        "text": "JMP",
        "url": "https://washdata.org/how-we-work/about-jmp#:~:text=Background,hygiene%20(WASH)%20since%201990",
      },
      Object {
        "text": "CLTS RTMIS",
        "url": " http://wash.health.go.ke/clts/index.jsp",
      },
      Object {
        "text": "GLAAS",
        "url": "https://www.who.int/teams/environment-climate-change-and-health/water-sanitation-and-health/monitoring-and-evidence/wash-systems-monitoring/un-water-global-analysis-and-assessment-of-sanitation-and-drinking-water",
      },
    ],
    "footerExternalLinkTitle": "External Links",
    "footerQuickLinkItems": Array [
      Object {
        "text": "Read the Docs",
        "url": "/documentation",
      },
    ],
    "footerQuickLinkTitle": "Quick Links",
    "footerResourcesItems": Array [
      Object {
        "text": "International Resources",
        "url": "#",
      },
      Object {
        "text": "National Resources",
        "url": "#",
      },
      Object {
        "text": "County Resources",
        "url": "#",
      },
    ],
    "footerResourcesTitle": "Resources",
    "forgotDesc": "Enter the email associated with your account and we&apos;ll Send an email with instructions to reset your password",
    "forgotTitle": "Reset your password",
    "formAssignmentError": "You don't have any form assignment, please contact the administrator",
    "formDescription": <p>
      Please fill up the webform below with relevant responses. You will need to answer all mandatory questions before you can submit.
      <br />
      Once you have sumitted a webform, please do not forget to add it as part of a batch and send it for approval.
    </p>,
    "formSuccessSubTitle": "Do note that this data has NOT been sent for approval. If you are ready to send the submissions for approval, please create a batch and send to the approver",
    "formSuccessSubTitleForAdmin": "Do note that the data submitted by SUPER ADMIN role will not go through the approval flow and recorded as approved data",
    "formSuccessTitle": "Thank you for the submission",
    "generating": "Generating",
    "hour": "Hour",
    "informUser": "Inform User for Changes",
    "instructionsMailed": "Instructions mailed successfully",
    "invalidInviteDesc": "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Autem provident voluptatum cum numquam, quidem vitae, qui quam beatae exercitationem ullam perferendis! Nobis in aut fuga voluptate harum, tempore distinctio optio.",
    "invalidInviteTitle": "Invalid Invite Code",
    "login": "Log in",
    "loginTitle": <React.Fragment>
      Welcome back
      <br />
      <small>
        Please enter your account details
      </small>
    </React.Fragment>,
    "lorem": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Possimus, assumenda quos? Quia deleniti sapiente aut! Ab consequatur cumque fugit ea. Dolore ex rerum quisquam inventore eum dicta doloribus harum cum.",
    "lorem2": "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
    "manageDataButton": "Manage Data",
    "manageDataText": <React.Fragment>
      This is where you :
      <ul>
        <li>
          Add new data using webforms
        </li>
        <li>
          Bulk upload data using spreadsheets
        </li>
        <li>
          Download data
        </li>
      </ul>
    </React.Fragment>,
    "manageDataTitle": "Manage Data",
    "manageOrganisations": "Manage Organizations",
    "manageQnApproval": "Manage Questionnaire Approval",
    "manageUserButton": "Manage Users",
    "manageUserText": <React.Fragment>
      This is where you manage users based on their roles , regions and questionnaire access . You can :
      <ul>
        <li>
          Add new user
        </li>
        <li>
          Modify existing user
        </li>
        <li>
          Delete existing user
        </li>
      </ul>
    </React.Fragment>,
    "manageUserTitle": "User Management",
    "manageUsers": "Manage Users",
    "minute": "Minute",
    "month": "Month",
    "myProfile": "My Profile",
    "newsEvents": "News & Events",
    "next": "Next",
    "notAssigned": "Not assigned",
    "notesFeedback": "Notes & Feedback",
    "orgPanelButton": "Manage Organization",
    "orgPanelText": <React.Fragment>
      This is where you :
      <ul>
        <li>
          Add new organization
        </li>
        <li>
          Modify existing organization
        </li>
        <li>
          Delete existing organization
        </li>
      </ul>
    </React.Fragment>,
    "orgPanelTitle": "Manage Organization",
    "organisationDeleteFail": "Could not delete organization",
    "organisationsLoadFail": "Could not load organizations",
    "panelApprovalsDesc": <React.Fragment>
      This is where you :
      <ul>
        <li>
          View pending data approvals awaiting your approval 
        </li>
        <li>
          View pending approvals by your subordinate approvers
        </li>
        <li>
          Assign subordinate approvers
        </li>
      </ul>
    </React.Fragment>,
    "passwordCriteriaError": "False Password Criteria",
    "passwordMatchError": "The two passwords that you entered do not match!",
    "passwordRequired": "Please input your Password!",
    "passwordRule1": "Lowercase Character",
    "passwordRule2": "Numbers",
    "passwordRule3": "Special Character ( -._!\`'#%&,:;<>=@{}~$()*+/?[]^|] )",
    "passwordRule4": "Uppercase Character",
    "passwordRule5": "No White Space",
    "passwordRule6": "Minimum 8 Characters",
    "passwordUpdateSuccess": "Password updated successfully",
    "prev": "Prev",
    "reports": "Reports",
    "resetHint": <React.Fragment>
      Please set your password for the platform.
      <br />
      Your password must include:
    </React.Fragment>,
    "second": "Second",
    "selectForm": "Please select a form",
    "sendInstructions": "Send Instructions",
    "sendNewRequest": "Notify Approver",
    "settings": "System Settings",
    "settingsDescriptionPanel": "This page allows Super Admin to maintain system critical master lists.",
    "showEmpty": "Show empty values",
    "signOut": "Sign Out",
    "submissionComment": "Submission comment",
    "templateDownloadHint": "If you do not already have a template, please download it",
    "templateFetchFail": "Could not fetch template",
    "tourApprovals": "Placeat impedit iure quaerat neque sit quasi",
    "tourApprovers": "Magni provident aliquam harum cupiditate iste",
    "tourControlCenter": "Lorem ipsum dolor sit, amet consectetur adipisicing elit",
    "tourDataUploads": "Velit amet omnis dolores. Ad eveniet ex beatae dolorum",
    "tourDataUploadsPanel": "Velit amet omnis dolores. Ad eveniet ex beatae dolorum",
    "tourExports": "Velit amet omnis dolores. Ad eveniet ex beatae dolorum",
    "tourManageData": "Lorem ipsum dolor sit, amet consectetur adipisicing elit",
    "tourUserManagement": "Magni provident aliquam harum cupiditate iste",
    "updateExisting": "Update Existing Data",
    "updateOrganisation": "Update Organization",
    "updateUser": "Update User",
    "uploadTab1": "Data Summary",
    "uploadTab2": "Raw Data",
    "uploading": "Uploading..",
    "uploadsTab1": "Pending Submission",
    "uploadsTab2": "Pending Approval",
    "uploadsTab3": "Approved",
    "userAssociations": "This user has following data association(s)",
    "userDeleteFail": "Could not delete user",
    "usernameRequired": "Please input your Username!",
    "usersLoadFail": "Could not load users",
    "valDesignation": "Please select a Designation",
    "valEmail": "Please enter a valid Email Address",
    "valFirstName": "First name is required",
    "valLastName": "Last name is required",
    "valOrgAttributes": "Please select an Attributes",
    "valOrgName": "Organization name is required",
    "valOrganization": "Please select an Organization",
    "valPhone": "Phone number is required",
    "valRole": "Please select a Role",
    "welcome": <React.Fragment>
      Kenya Rural Urban Sanitation
      <br />
      and Hygiene (RUSH)
    </React.Fragment>,
    "welcomeCta": "Explore National Data",
    "welcomeDesc": <React.Fragment>
      Real Time Management Information System (RTMIS)
      <br />
      Updated sanitation and hygiene estimates across Kenya
    </React.Fragment>,
    "welcomeShort": "Welcome to RUSH",
    "year": "Year",
  },
}
`;
